Newsletters: add 159 (2021-07-28)

harding opened the pull request:
- [x] Add releases/RCs @harding
- [x] Add topics @harding 
- [x] Bitcoin Stack Exchange section @bitschmidty 
- [x] `C-Lightning #4669` Updates to offers @dongcarl 
- [x] `BIPs #1072` BIP48 HD Multisig derivation standard @Xekyo 

Note: the entry about dual-funding/liquidity advertisements support for "sidecar channels" is currently unverified.  I think my reasoning about it is sound, but I asked for confirmation in `#lightning-dev` IRC.

jnewbery reviewed (commented):
Looks good to me. I don't know enough about liquidity advertisements to do a technical review of the c-lightning 4639 write-up.

jnewbery commented on _posts/en/newsletters/2021-07-28-newsletter.md:
```diff
@@ -67,7 +67,8 @@ BOLTs][bolts repo].*
 
-- [C-Lightning #4669][] Updates to offers FIXME:dongcarl
+- [C-Lightning #4669][] fixes several bugs in its [LN offers][topic offers]
+  parsing and validation logic reported by Nadav Igvi.
```
We generally don't attribute individuals in the "notable code changes" section, since the only fair thing to do would be to attribute all authors, reviewers, testers, etc.

```suggestion
  parsing and validation logic.
```

Xekyo commented on _posts/en/newsletters/2021-07-28-newsletter.md:
```diff
@@ -0,0 +1,183 @@
+- [What is this unusual transaction in the Bitcoin blockchain?]({{bse}}107603)
+  Murch describes an output labeled "UNKNOWN" in a [block explorer][topic
+  block explorers]. The output is a segwit version 1 output with a contrived
+  pubkey. As pointed out 0xb10c, the 2019 transaction creating this output was for
+  purposes of testing segwit v1 support for [Optech's Compatibility
```
```suggestion
  pubkey. As pointed out 0xb10c, the 2019 transaction creating this output was for
  the purpose of testing segwit v1 support for [Optech's Compatibility
```

Xekyo commented on _posts/en/newsletters/2021-07-28-newsletter.md:
```diff
@@ -0,0 +1,183 @@
+- [What is this unusual transaction in the Bitcoin blockchain?]({{bse}}107603)
+  Murch describes an output labeled "UNKNOWN" in a [block explorer][topic
+  block explorers]. The output is a segwit version 1 output with a contrived
+  pubkey. As pointed out 0xb10c, the 2019 transaction creating this output was for
```
```suggestion
  pubkey. As pointed out by 0xb10c, the 2019 transaction creating this output was for
```

Xekyo commented on _posts/en/newsletters/2021-07-28-newsletter.md:
```diff
@@ -0,0 +1,183 @@
+  block explorers]. The output is a segwit version 1 output with a contrived
+  pubkey. As pointed out 0xb10c, the 2019 transaction creating this output was for
+  purposes of testing segwit v1 support for [Optech's Compatibility
+  Matrix][compat matrix]. As warned previously (see [Newsletter #158][news158
+  taproot]), 0xb10c also inspects some [anyone-can-spend P2TR outputs][0xB10C blog].
```
This sounds as if we warned that 0xb10c would inspect P2TR outputs, but we warned that anyone could spend them before the activation of taproot.

```suggestion
  Matrix][compat matrix]. As warned previously (see [Newsletter #158][news158
  taproot]), P2TR outputs are anyone-can-spend before the activation of taproot as 0xb10c demonstrated and [elaborates in a blog post][0xB10C blog].
```

Xekyo commented on _posts/en/newsletters/2021-07-28-newsletter.md:
```diff
@@ -0,0 +1,183 @@
+
+- [What are miners signalling for when the block header nversion field ends in 4 i.e. 0x3fffe004?]({{bse}}107443)
+  While researching the overt form of ASICBoost, user shikaridota wonders why
+  recently mined blocks have bit 2 being set in the `nVersion` field. Andrew
+  Chow points out that for [taproot][topic taproot], [BIP341's
+  deployment][bip341 deployment] section specifies bit 2 for activation.
```
```suggestion
  recently mined blocks have bit 2 being set in the `nVersion` field. Andrew
  Chow points out that [taproot][topic taproot] used bit 2 to signal for activation as specified in [BIP341's
  deployment][bip341 deployment] section.
```

Xekyo commented on _posts/en/newsletters/2021-07-28-newsletter.md:
```diff
@@ -0,0 +1,183 @@
+answers posted since our last update.*
+
+{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
+{% assign bse = "https://bitcoin.stackexchange.com/a/" %}
```
Would it perhaps make sense to prescribe this alias site-wide?

harding replied:
I think it's @bitschmidty's call, since he's using this.  It would change the variable name from `{{bse}}` to `{{site.bse}}` (or `site.whatever_mike_wants`), which might be a tiny bit more effort to type.

Xekyo commented on _posts/en/newsletters/2021-07-28-newsletter.md:
```diff
@@ -0,0 +1,183 @@
+
+   </table>
+
+- [BIPs #1072][] merges the informational BIP48 titled "Multi-Script Hierarchy
```
Let's link to the BIP here.

```suggestion
- [BIPs #1072][] merges the informational [BIP48][] titled "Multi-Script Hierarchy
```

Xekyo replied:
Done, thanks for the suggestion.

harding commented on _includes/specials/taproot/en/05-taproot-notebooks.md:
```diff
@@ -0,0 +1,55 @@
+
+- **Section 2** gives you experience with every aspect of taproot.  It
+  starts with a review of the principles of segwit v0 transactions and then
+  helps you create and spend segwit v1 (taproot) transactions.  Applying
```
Instead of this I did s/spend/send/ since the section being described also includes other bits about serializing the transaction and creating the signature hash.

harding commented:
Made requested edits (thanks!), reviewed other contributors' content (more thanks!), added releases/rcs and topic links, and changed the entry about liquidity advertisements compatibility with sidecar channels to link to a StackExchange question I created since I was unable to get a reply about it in IRC.

Thanks everyone!

bitschmidty commented:
Great job everyone! Squashed and merged. TY @harding @dongcarl @Xekyo and @jnewbery !

jnewbery commented:
Sorry I didn't review the topics and SE before merge, but I've taken a look now and all looks good to me. Thanks @bitschmidty

bitschmidty commented:
@jnewbery No problem. I was a bit later that I would have liked.
